app/battle/battlefront.py

```diff
@@ -0,0 +1,31 @@
+from app.battle.knight import Knight
+
+
+def battle(knights_config: dict) -> dict:
+    lancelot = Knight(**knights_config["lancelot"])
+    arthur = Knight(**knights_config["arthur"])
+    mordred = Knight(**knights_config["mordred"])
+    red_knight = Knight(**knights_config["red_knight"])
+
+    battle1 = fight(lancelot, mordred)
+    battle2 = fight(arthur, red_knight)
```

Comment:
Consider encapsulating the logic of reducing `hp` within the `Knight` class itself. This would make the `fight` function cleaner and adhere to object-oriented principles by allowing each `Knight` to manage its own state.